File: 0496-next-greater-element-i/0496-next-greater-element-i.cpp
class Solution {
public:
    vector<int> nextGreaterElement(vector<int>& nums1, vector<int>& nums2) {
        vector<int> v(nums1.size(),-1);
        for(int i=0;i<nums1.size();i++)
        {
            int pos=find(nums2.begin(),nums2.end(),nums1[i])-nums2.begin();
            while(pos<nums2.size())
            {
                if(nums2[pos]>nums1[i])
                {
                v[i]=nums2[pos];
                break;}
                pos++;
            }
            cout<<endl;
        }
        return v;
    }
};

File: 0496-next-greater-element-i/0496-next-greater-element-i.java
class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
        Stack<Integer> st=new Stack<>();
        int n=nums2.length;
        HashMap<Integer,Integer> mp =new HashMap<>();
        for(int i=n-1;i>=0;i--)
        {
            while(!st.empty() && nums2[i]>st.peek())
            {
                st.pop();
            }
            if(st.empty())
            {
                mp.put(nums2[i],-1);
            }
            else
            {
            mp.put(nums2[i],st.peek());
            }
            st.push(nums2[i]);
        }
        n=nums1.length;
        int p=0;
        int ar[]=new int[n];
        for(int i:nums1)
        {
            ar[p++]=mp.get(i);
        }
        return ar;
    }
}